framework/CodeInterfaces/SERPENT/SerpentInterface.py

```diff
@@ -0,0 +1,191 @@
+      @ In, inputFiles, list, List of input files (lenght of the list depends on the number of inputs have been added in the Step is running this code)
+      @ In, executable, string, executable name with absolute path (e.g. /home/path_to_executable/code.exe)
+      @ In, clargs, dict, optional, dictionary containing the command-line flags the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 i0extension =0 .inp0/ >< /Code >)
+      @ In, fargs, dict, optional, a dictionary containing the axuiliary input file variables the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 aux0extension =0 .aux0/ >< /Code >)
+      @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is the command to run the code (string), returnCommand[1] is the name of the output root
+    """
+    if clargs==None:
+      raise IOError('No input file was specified in clargs!')
+    #check for output either in clargs or fargs
+    #if len(fargs['output'])<1 and 'output' not in clargs.keys():
+    #  raise IOError('No output file was specified, either in clargs or fileargs!')
+    #check for duplicate extension use
+    usedExt=[]
+    for ext in list(clargs['input'][flag] for flag in clargs['input'].keys()) + list(fargs['input'][var] for var in fargs['input'].keys()):
+      if ext not in usedExt:
+        usedExt.append(ext)
+      else:
+        raise IOError('SERPENT cannot handle multiple input files with the same extension.  You may need to write your own interface.')
+
+    #check all required input files are there
+    inFiles=inputFiles[:]
+    for exts in list(clargs['input'][flag] for flag in clargs['input'].keys()) + list(fargs['input'][var] for var in fargs['input'].keys()):
+      for ext in exts:
+        found=False
+        for inf in inputFiles:
+          if '.'+inf.getExt() == ext:
+            found=True
+            inFiles.remove(inf)
+            break
+        if not found:
+          raise IOError('input extension "'+ext+'" listed in input but not in inputFiles!')
+    #TODO if any remaining, check them against valid inputs
+
+    #PROBLEM this is limited, since we can't figure out which .xml goes to -i and which to -d, for example.
+    def getFileWithExtension(fileList,ext):
+      """
+      Just a script to get the file with extension ext from the fileList.
+      @ In, fileList, the string list of filenames to pick from.
+      @ Out, ext, the string extension that the desired filename ends with.
+      """
+      found = False
+      for index,inputFile in enumerate(fileList):
+        if inputFile.getExt() == ext:
+          found=True
+          break
+      if not found:
+        raise IOError('No InputFile with extension '+ext+' found!')
+      return index,inputFile
+
+    #prepend
+    todo = ''
+    todo += clargs['pre']+' '
+    todo += executable
+    index=None
+    #inputs
+    for flag,exts in clargs['input'].items():
```

Comment:
This looks copied/pasted from the GenericCode Interface. Do you need all these features for running Serpent?

Reply:
(I mean all the code  from ```def getFileWithExtension(fileList,ext)``` to ```return returnCommand```